rust/agent-anthropic/src/lib.rs

```diff
@@ -0,0 +1,98 @@
+use std::{env, sync::Arc};
+
+use anthropic::{
+    client::Client,
+    config::AnthropicConfig,
+    types::CompleteRequestBuilder,
+    AI_PROMPT, HUMAN_PROMPT,
+};
+
+
+use agent::{
+    common::{
+        async_trait::async_trait,
+        eyre::{bail, Result},
+    },
+    Agent, AgentIO, GenerateOptions,
+};
+
+/// An agent running on OpenAI
+///
+/// The environment variable OPENAI_API_KEY must be set to use these agents.
+pub struct AnthropicAgent {
+    /// The OpenAI name for a model including any tag e.g. "llama2:13b"
+    ///
+    /// Used as the required `model` parameter in each request to `POST /api/generate`
+    /// (along with `prompt`).
+    model: String,
+
+}
+
+impl AnthropicAgent {
+    /// Create a OpenAI-based agent
+    pub fn new(model: String) -> Self {
+        Self {
+            model,
+        }
+    }
+   fn supported_inputs(&self) -> &[AgentIO] {
+        &[AgentIO::Text]
+    }
+
+    fn supported_outputs(&self) -> &[AgentIO] {
+        &[AgentIO::Text]
+    }
+}
+
+#[async_trait]
+impl Agent for AnthropicAgent {
+
+    fn provider(&self) -> String {
+        "anthropic".to_string()
+    }
+
+    fn model(&self) -> String {
+        self.model.clone()
+    }
+
+    // fn supported_inputs(&self) -> &[AgentIO] {
+    //     &[AgentIO::Text]
+    // }
+
+    // fn supported_outputs(&self) -> &[AgentIO] {
+    //     &[AgentIO::Text]
+    // }
+
+    async fn text_to_text(&self, instruction: &str, options: &GenerateOptions) -> Result<String> {
+        // Build from configuration.
+        // TODO: We need to use the configuration here.
+        // Currently just a cut/paste from the example.
+        let cfg = AnthropicConfig::new()?;
+        let client = Client::try_from(cfg)?;
+
+        let complete_request = CompleteRequestBuilder::default()
+            .prompt(format!("{HUMAN_PROMPT}{instruction}{AI_PROMPT}"))
+            // What is claude-instant-1??
+            .model("claude-instant-1".to_string())
+            .max_tokens_to_sample(256usize)
```

Comment:
This should really come from options, rather than be fixed. One option would be to use a fallback e.g

```suggestion
            .max_tokens_to_sample(options.max_tokens.unwrap_or(256) as usize)
```

Other options like `temperature` and `seed` (if available) should also be set or log a warning if they can't be set

Reply:
* The max tokens is done. 
* The other options cannot be set, as they are not included in the rust crate. We could do a simple PR to add them (and fork for now, as we already need to use the git repo directly)
* I haven't dealt with logging warnings yet.